src/genopets/infos.ts

```diff
@@ -0,0 +1,233 @@
+  DataSizeFilter,
+  MemcmpFilter,
+  GetProgramAccountsConfig,
+} from "@solana/web3.js";
+import { IFarmInfoWrapper, IInstanceFarm } from "../types";
+import { GENOPETS_FARM_PROGRAM_ID } from "./ids";
+import { DEPOSIT_LAYOUT, FARMER_LAYOUT, FARM_LAYOUT } from "./layouts";
+import * as types from ".";
+import { BN } from "bn.js";
+import { getMultipleAccounts } from "../utils";
+
+let infos: IInstanceFarm;
+infos = class InstanceGenopets {
+  static async getAllFarms(connection: Connection): Promise<types.FarmInfo[]> {
+    const farmId = PublicKey.findProgramAddressSync([Buffer.from("stake-master-seed")], GENOPETS_FARM_PROGRAM_ID)[0];
+    const farmAccountInfo = await connection.getAccountInfo(farmId);
+    const farm = this.parseFarm(farmAccountInfo?.data!, farmId);
+
+    return [farm];
+  }
+
+  static async getAllFarmWrappers(connection: Connection): Promise<types.FarmInfoWrapper[]> {
+    return (await this.getAllFarms(connection)).map((farmInfo) => new FarmInfoWrapper(farmInfo));
+  }
+
+  static async getFarm(connection: Connection, farmId: PublicKey): Promise<types.FarmInfo> {
+    const expectedFarmId = PublicKey.findProgramAddressSync(
+      [Buffer.from("stake-master-seed")],
+      GENOPETS_FARM_PROGRAM_ID
+    )[0];
+    if (!farmId.equals(expectedFarmId)) throw "Error: Wrong farmId.";
+    let account = (await connection.getAccountInfo(farmId)) as AccountInfo<Buffer>;
+    return this.parseFarm(account.data, farmId);
+  }
+
+  static async getFarmWrapper(connection: Connection, farmId: PublicKey): Promise<FarmInfoWrapper> {
+    const farm = await this.getFarm(connection, farmId);
+
+    return new FarmInfoWrapper(farm);
+  }
+
+  static parseFarm(data: Buffer, farmId: PublicKey): types.FarmInfo {
+    const decodedData = FARM_LAYOUT.decode(data);
+
+    let {
+      authority,
+      sgeneMinter,
+      mintSgene,
+      geneMint,
+      geneRewarder,
+      totalGeneRewarded,
+      ataGeneRewarder,
+      totalGeneAllocated,
+      totalWeight,
+      startTime,
+      endTime,
+      epochTime,
+      decayFactorPerEpoch,
+      initialGenesPerEpoch,
+      stakeParams,
+      pausedState,
+      totalRewardWeight,
+      accumulatedYieldRewardsPerWeight,
+      lastYieldDistribution,
+      totalGeneStaked,
+      timeFactor,
+    } = decodedData;
+
+    return {
+      farmId,
+      authority,
+      sgeneMinter,
+      mintSgene,
+      geneMint,
+      geneRewarder,
+      totalGeneRewarded,
+      ataGeneRewarder,
+      totalGeneAllocated,
+      totalWeight,
+      startTime,
+      endTime,
+      epochTime,
+      decayFactorPerEpoch,
+      initialGenesPerEpoch,
+      stakeParams,
+      pausedState,
+      totalRewardWeight,
+      accumulatedYieldRewardsPerWeight,
+      lastYieldDistribution,
+      totalGeneStaked,
+      timeFactor,
+    };
+  }
+
+  static async getAllFarmers(connection: Connection, userKey: PublicKey): Promise<types.FarmerInfo[]> {
+    const farmerId = PublicKey.findProgramAddressSync(
```

Comment:
Use `getFarmerId`

Reply:
`getFarmerId` required `farmInfo` as one of the input param, but we won't have it here.

Reply:
k, lets keep it then